common/setups/rasr/nn_system.py

```diff
@@ -0,0 +1,564 @@
+                network[forward_output_layer]["activation"] = "log_sigmoid"
+            elif network[forward_output_layer]["activation"] == "exp":
+                network[forward_output_layer]["activation"] = None
+            elif network[forward_output_layer]["activation"] is None:
+                network[forward_output_layer]["activation"] = "log"
+        # target = 'classes'
+        if "cropped" in network:
+            if network["output"]["from"] == ["cropped"]:
+                network["output"]["from"] = "upsample"
+            network.pop("cropped")
+            if "lstm_bwd_1" in network:
+                network["lstm_bwd_1"]["from"] = "upsample"
+                network["lstm_fwd_1"]["from"] = "upsample"
+            if "lstm_fwd_1_no_init" in network:
+                network["lstm_bwd_1_no_init"]["from"] = "upsample"
+                network["lstm_fwd_1_no_init"]["from"] = "upsample"
+
+        return config
+
+    @staticmethod
+    def get_tf_flow(
+        checkpoint_path: Path,
+        tf_graph_path: Path,
+        returnn_op_path: Path,
+        forward_output_layer: str = "output",
+        tf_fwd_input_name: str = "tf-fwd-input",
+    ):
+        """
+        Create flow network and config for the tf-fwd node
+
+        :param Path checkpoint_path: RETURNN model checkpoint which should be loaded
+        :param Path tf_graph_path: compiled tf graph for the model
+        :param Path returnn_op_path: path to native lstm library
+        :param str forward_output_layer: name of layer whose output is used
+        :param str tf_fwd_input_name: tf flow node input name. see: add_tf_flow_base_flow()
+        :rtype: FlowNetwork
+        """
+        input_name = tf_fwd_input_name
+
+        tf_flow = rasr.FlowNetwork()
+        tf_flow.add_input(input_name)
+        tf_flow.add_output("features")
+        tf_flow.add_param("id")
+        tf_fwd = tf_flow.add_node("tensorflow-forward", "tf-fwd", {"id": "$(id)"})
+        tf_flow.link(f"network:{input_name}", tf_fwd + ":input")
+        tf_flow.link(tf_fwd + ":log-posteriors", "network:features")
+
+        tf_flow.config = rasr.RasrConfig()
+
+        tf_flow.config[tf_fwd].input_map.info_0.param_name = "input"
+        tf_flow.config[
+            tf_fwd
+        ].input_map.info_0.tensor_name = "extern_data/placeholders/data/data"
+        tf_flow.config[
+            tf_fwd
+        ].input_map.info_0.seq_length_tensor_name = (
+            "extern_data/placeholders/data/data_dim0_size"
+        )
+
+        tf_flow.config[tf_fwd].output_map.info_0.param_name = "log-posteriors"
+        tf_flow.config[
+            tf_fwd
+        ].output_map.info_0.tensor_name = f"{forward_output_layer}/output_batch_major"
+
+        tf_flow.config[tf_fwd].loader.type = "meta"
+        tf_flow.config[tf_fwd].loader.meta_graph_file = tf_graph_path
+        tf_flow.config[tf_fwd].loader.saved_model_file = checkpoint_path
```

Comment:
Are you sure this worked for you? I got some problems when directly passing the checkpoint, as the path had the `*.index` extension which RASR did not accept, so I needed to add a "delayed cutting" function to solve this.

Reply:
I think this was the problem with `Model` and we specifically designed `Checkpoint` such that this works..
Here (https://github.com/rwth-i6/i6_core/blob/deb2bbe5748e9f9e4e9c6740290928beb5f1edf9/returnn/training.py#L55) the additional `".index"` should be removed.

Reply:
>  I will set `ReturnnModel` to deprecated

I don't think I meant `ReturnnModel` (which contains a path, a config, and an epoch) but a different class that was effectively doing the same as `Checkpoint` but was hashing the absolute path to the model. (And was already removed)

> The `ckpt_path` can not be used here, because this already calls get_path()

But we would only set the `Checkpoint` object here. The call to `Checkpoint.str` (and subsequently `ckpt_path`) would only happen at runtime when the `create_files` Task of Recognition was executed; Therefore we _can_ use it here, right?